app/assets/javascripts/miq_application.js

```diff
@@ -1482,6 +1482,11 @@ function miqToolbarOnClick(e) {
         }
       }
     }
+  } else if (button.data('function')) {
+    // support data-function and data-function-data
+    var fn = new Function("return " + button.data('function')); // eval - returns a function returning the right function
```

Comment:
yes, @zeari, that is about adding special behavior for those. Other thing is what other stuff will creative UI developers add if we allow anything to be passed.